src/utils/addNetworkPrefix.ts

```diff
@@ -0,0 +1,21 @@
+import { extractShortChainName } from 'src/routes/routes'
+import getNetworkPrefix from './getNetworkPrefix'
+
+function addNetworkPrefix(address = '', showNetworkPrefix = true, netWorkPrefix = extractShortChainName()): string {
+  // if showNetworkPrefix is disabled in appearance settings we return the address as it is
+  if (!showNetworkPrefix) {
+    return address
+  }
+
+  const prefix = getNetworkPrefix(address)
+
+  // if no prefix is present in the provided address we return the prefixed address
+  if (!prefix && address) {
+    const prefixedAddress = `${netWorkPrefix}:${address}`
```

Comment:
You can use `getPrefixedSafeAddressSlug()` for this.

Reply:
Are you sure about this? it is specific for the Safe Addresses that are present in the Url. In that function Prefix argument is a `enum SHORT_NAME`, but in the input users can type another values for the prefix.

Reply:
Why do you want to allow invalid (currently unsupported in the Safe) `shortNames`?

Reply:
It is needed to trigger the network prefix validation in the form.
If the user types an address like: `vt:0x57CB13cbef735FbDD65f5f2866638c546464E45F` in `Rinkeby`  we need to set this invalid value in the form state to make sure that the form is aware that a network prefix error  is present in that address field.
![Captura de pantalla 2021-11-08 a las 15 46 17](https://user-images.githubusercontent.com/26763673/140763004-9ba109d4-d1c3-4449-8450-df29b9332775.png)